docs/en/developers/references/traits/rollout.md

```diff
@@ -26,3 +26,29 @@ Name | Description | Type | Required | Default
  interval | Schedule interval time | string | true | 30s 
  stepWeight | Weight percent of every step in rolling update | int | true | 50 
  replicas | Total replicas of the workload | int | true | 2 
+
+## Conflicts With
+
+### `Autoscale`
+
+When `Rollout` and `Autoscle` traits are attached to the same service, they two will fight over the number of instances during rollout. Thus, it's by design that `Rollout` will take over replicas control (specified by `.replicas` field) during rollout.
+
+> Note: in up coming releases, KubeVela will introduce a separate section in Appfile to define release phase configurations such as `Rollout`.
+
+## How `Rollout` works?
+
+`Rollout` trait implements progressive release process to rollout your app following [Canary strategy](https://martinfowler.com/bliki/CanaryRelease.html).
+
+In detail, `Rollout` controller will create a canary of your app , and then gradually shift traffic to the canary while measuring key performance indicators like HTTP requests success rate at the same time. 
+
+
+![alt](../../../../../docs/resources/traffic-shifting-analysis.png)
+
+In this sample, for every `10s`, `5%` traffic will be shifted to canary from the primary, until the traffic on canary reached `50%`. At the mean time, the instance number of canary will automatically scale to `replicas: 2` per configured in Appfile.
+
+
+Based on analysis result of the KPIs during this traffic shifting, a canary will be promoted or aborted if analysis is failed. If promoting, the primary will be upgraded from v1 to v2, and traffic will be fully shifted back to the primary instances. So as result, canary instances will be deleted after the promotion finished.
+
+![alt](../../../../../docs/resources/promotion.png)
+
+> Note: KubeVela's `Rollout` trait is implemented with [Weaveworks Flagger](https://flagger.app/) operator.
```

Comment:
We are not implementing the rollout trait based on flagger.

Reply:
@wonderflow Please help review Ryan' comments. Thanks.

Reply:
Yes, @ryanzhang-oss  is implementing a new rollout #787,  we can hold this part.

Reply:
This is end user facing documentation regarding to current status, it's exactly Fagger until the new AppDeployment is released. We should not hold this.

Reply:
alright, lgtm and approved